src/Microsoft.Health.Fhir.SqlServer/Registration/FhirServerBuilderSqlServerRegistrationExtensions.cs

```diff
@@ -239,14 +238,13 @@ public static IFhirServerBuilder AddSqlServer(this IFhirServerBuilder fhirServer
                             .Singleton()
                             .AsSelf();
 
-            services
-                .RemoveServiceTypeExact<DefragWatchdog, INotificationHandler<StorageInitializedNotification>>()
-                .Add<DefragWatchdog>()
-                .Singleton()
-                .AsSelf()
-                .AsService<INotificationHandler<StorageInitializedNotification>>();
+            services.Add<DefragWatchdog>().Singleton().AsSelf();
+
+            services.Add<CleanupEventLogWatchdog>().Singleton().AsSelf();
 
-            services.AddHostedService<WatchdogsBackgroundService>();
+            services.AddSingleton<WatchdogsBackgroundService>();
+            services.AddHostedService(isp => isp.GetRequiredService<WatchdogsBackgroundService>());
+            services.AddSingleton<INotificationHandler<StorageInitializedNotification>>(isp => isp.GetRequiredService<WatchdogsBackgroundService>());
```

Comment:
We should register jobs the way all other jobs are registered.

Reply:
Did you check in your extension method changes?

Reply:
```csharp
  services
       .RemoveServiceTypeExact<YOUR_BACKGROUND_SERVICE, INotificationHandler<StorageInitializedNotification>>() // Mediatr registers handlers as Transient by default, this extension ensures these aren't still there, only needed when service != Transient
       .Add<YOUR_BACKGROUND_SERVICE>()
       .Singleton()
       .AsSelf()  // this is needed to create the instance the delegates resolve
       .AsService<IHostedService>()
       .AsService<INotificationHandler<StorageInitializedNotification>>();
```

Reply:
Fixed.